test/Datadog.Trace.Tests/SpanTests.cs

```diff
@@ -42,6 +42,97 @@ public void SetTag_KeyValue_KeyValueSet()
             Assert.Equal(span.GetTag(key), value);
         }
 
+        [Fact]
+        public void SetTag_KeyValue_DoubleValueSet()
+        {
+            const string key = "Key";
+            const double value = 123456789d;
+            var span = _tracer.StartSpan("Operation");
+            Assert.Null(span.GetTagValue(key));
+
+            span.SetTagValue(key, value);
+
+            _writerMock.Verify(x => x.WriteTrace(It.IsAny<Span[]>()), Times.Never);
+            Assert.Equal(value, span.GetTagValue(key).Value.DoubleValue);
+            Assert.Equal(value, (double)span.GetTagValue(key));
+        }
+
+        [Fact]
+        public void SetTag_Key_Value_DoubleValueOverflowSet()
+        {
+            const string key = "Key";
+            const double value = 9007199254740992;
```

Comment:
NIT: You should expose Pow2_53 as an internal const and use it in the test. It gives more context as to where the value comes from

Reply:
done